src/component/plotly-dash-preview/parse.js

```diff
@@ -0,0 +1,39 @@
+const isNonEmptyString = require('../../util/is-non-empty-string')
+
+/**
+ * @param {object} body : JSON-parsed request body
+ *  - url
+ *  - share_key
+ *  - parameters
+ * @param {object} opts : component options
+ * @param {function} sendToRenderer
+ * - errorCode
+ * - result
+ */
+function parse (body, opts, sendToRenderer) {
+  const result = {}
+
+  const errorOut = (code, msg) => {
+    result.msg = msg
+    sendToRenderer(code, result)
+  }
+
+  if (isNonEmptyString(body.url)) {
+    result.url = body.url
+  } else {
+    return errorOut(400, 'invalid url')
+  }
+
+  if (isNonEmptyString(body.format) && (body.format === 'png' || body.format === 'pdf')) {
+    result.format = body.format
+  } else {
+    // Default
+    result.format = 'png'
+  }
+
+  result.width = body.width || 800
+  result.height = body.height || 600
```

Comment:
This is interesting, I didn't think about this. I'm assuming that with PDF reports, where the content has absolute pixel sizing (e.g. the pages are 8.5" wide and 11" long), it doesn't matter what width and height we provide. However, for responsive apps, we'll need to provide something reasonable. Let's try a medium sized laptop: `width: 1680px`, `height: 1050`

Reply:
These options are only meant for `png` generation, as the printToPDF outputs are determined by: https://github.com/electron/electron/blob/master/docs/api/web-contents.md#contentsprinttopdfoptions-callback .. and not be the browser-size.

Although, we _can_ add them in printToPDF options after some conversion as `printToPDF` expects height/width in microns.

Reply:
Ah I understand now. OK, lets just use the electron options directly instead of doing a pixel to micron conversion.